syntax = "proto3";

package modelservice;

service ModelService {
  rpc ListModels (Empty) returns (ModelList);
  rpc TrainModel (TrainRequest) returns (TrainResponse);
  rpc Predict (PredictRequest) returns (PredictResponse);
  rpc DeleteModel (DeleteRequest) returns (DeleteResponse);
}

message Empty {}

message ModelList {
  repeated string models = 1;
}

message TrainRequest {
  string model_type = 1;
  repeated double X = 2 [packed = true];
  repeated int32 y = 3 [packed = true];
}

message TrainResponse {
  string message = 1;
  string model_id = 2;
}

message PredictRequest {
  string model_id = 1;
  repeated double X = 2 [packed = true];
}

message PredictResponse {
  repeated double predictions = 1;
}

message DeleteRequest {
  string model_id = 1;
}

message DeleteResponse {
  string message = 1;
}
